source/source.go

```diff
@@ -107,12 +107,12 @@ func (s *Source) Read(ctx context.Context) (sdk.Record, error) {
 	sqsMessagesC := make(chan *sqs.ReceiveMessageOutput)
 	errC := make(chan error)
 	go func() {
```

Comment:
Why do we need a go function here at all? That's really strange, I'd just call `s.svc.ReceiveMessage` directly, or does it not respect context cancellation?

Reply:
Oh sorry! I might have completely overlooked the context parameter, let me fix that.